apps/frontend/src/app/5_pages/StakePage/components/PersonalStakingStatistics/hooks/useGetPersonalStakingStatistics.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useAccount } from '../../../../../../hooks/useAccount';
+import { useGetProtocolContract } from '../../../../../../hooks/useGetContract';
+import { asyncCall } from '../../../../../../store/rxjs/provider-cache';
+import { getRskChainId } from '../../../../../../utils/chain';
+import { useGetStakingBalanceOf } from '../../../hooks/useGetStakingBalanceOf';
+
+export const useGetPersonalStakingStatistics = () => {
+  const { account } = useAccount();
+  const stakingContract = useGetProtocolContract('staking', getRskChainId());
+  const [votingPower, setVotingPower] = useState(0);
+  const [isDataFetched, setIsDataFetched] = useState(false);
+
+  const { balance } = useGetStakingBalanceOf(account);
+
+  const updateVotingPower = useCallback(async () => {
+    if (!stakingContract || isDataFetched) return;
+
+    try {
+      const result = await asyncCall('votingPower', () =>
```

Comment:
first param is id and is shared across dapp - make sure it has some uniqueness like account for which you are retrieving data (otherwise if you would connect another account you would see voting power of previous account)
 
```suggestion
      const result = await asyncCall(`${account}/votingPower` () =>
```